staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/streaming.go

```diff
@@ -0,0 +1,242 @@
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field")
+	}
+	typeMeta, ok := typeMetaValue.Interface().(metav1.TypeMeta)
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field to have TypeMeta type")
+	}
+
+	itemsField, ok := listType.FieldByName("Items")
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected Items field")
+	}
+	if itemsField.Tag.Get("json") != "items" {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field tag")
+	}
+	items, err := meta.ExtractList(list)
+	if err != nil {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, err
+	}
+	return typeMeta, listMeta, items, nil
+}
+
+func streamingEncodeList(w io.Writer, typeMeta metav1.TypeMeta, listMeta metav1.ListMeta, items []runtime.Object) error {
+	// Start
+	if _, err := w.Write([]byte(`{`)); err != nil {
+		return err
+	}
+
+	// TypeMeta
+	if typeMeta.Kind != "" {
+		if err := encodeKeyValuePair(w, "kind", typeMeta.Kind, true); err != nil {
+			return err
+		}
+	}
+	if typeMeta.APIVersion != "" {
+		if err := encodeKeyValuePair(w, "apiVersion", typeMeta.APIVersion, true); err != nil {
+			return err
+		}
+	}
+
+	// ListMeta
+	if err := encodeKeyValuePair(w, "metadata", listMeta, true); err != nil {
+		return err
+	}
+
+	// Items
+	if err := encodeItemsObjectSlice(w, items, false); err != nil {
+		return err
+	}
+
+	// End
+	_, err := w.Write([]byte("}\n"))
+	return err
+}
+
+func encodeItemsObjectSlice(w io.Writer, items []runtime.Object, sep bool) (err error) {
+	if items == nil {
+		err := encodeKeyValuePair(w, "items", nil, sep)
+		return err
+	}
+	_, err = w.Write([]byte(`"items":[`))
+	if err != nil {
+		return err
+	}
+	for i, item := range items {
+		err := encodeValue(w, item, i != len(items)-1)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = w.Write([]byte("]"))
+	if err != nil {
+		return err
+	}
+	if sep {
+		_, err = w.Write([]byte(","))
+	}
+	return err
+}
+
+func streamingEncodeUnstructuredList(w io.Writer, list *unstructured.UnstructuredList) error {
+	_, err := w.Write([]byte(`{`))
+	if err != nil {
+		return err
+	}
+	keys := slices.Sorted(maps.Keys(list.Object))
+	encodedItems := false
+	for i, key := range keys {
+		value := list.Object[key]
+		if key == "items" || (!encodedItems && key > "items") {
+			err := encodeItemsUnstructuredSlice(w, list.Items, i != len(keys)-1 || key != "items")
+			if err != nil {
+				return err
+			}
+			encodedItems = true
+			if key == "items" {
+				continue
+			}
+		}
+		err := encodeKeyValuePair(w, key, value, i != len(keys)-1 || !encodedItems)
+		if err != nil {
+			return err
+		}
+	}
+	if !encodedItems {
+		err := encodeItemsUnstructuredSlice(w, list.Items, false)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = w.Write([]byte("}\n"))
+	return err
+}
+
+func encodeItemsUnstructuredSlice(w io.Writer, items []unstructured.Unstructured, sep bool) (err error) {
```

Comment:
same comment about taking `suffix []byte` instead of `sep bool`, and doing this:

```go
	if len(suffix) > 0 {
		_, err = w.Write(suffix)
	}
```

"suffix" as a parameter is easier to understand what it will be used for (gets put at the end of the output) and lets the functions be used in various contexts more confidently

Reply:
Done